### ISSUE TITLE
pep9term documentation request.

### ISSUE THREAD

**Issue by StanWarford** [StanWarford/pep9suite#102]
Simplify and tighten up the output with `pep9term --help` from
```
Name:
pep9term -- Translate and run Pep/9 assembly language and microcode programs.

Usage:
pep9term [--help] [--help-all] [--version] [--about] 
pep9term asm [--help] [-e error_file] -s source_file -o object_file 
pep9term run [--help] [-i charin_file] [-o charout_file] [--echo-output] [-m max_steps] -s object_file 
pep9term cpuasm [--help] [-e error_file] [--d2] -s microcode_file 
pep9term cpurun [--help] [-e error_file] [--d2] [-p precondition_file] -s microcode_file 


Flags:
  -h,--help                   Show this help information.
  --help-all                  Show help information for all subcommands.
  -v,--version                Display program version number.
  --about                     Display information about licensing, Qt, and developers.

Subcommands:
  asm                         Assemble a Pep/9 assembler source code program to object code.
  run                         Run a Pep/9 object code program.
  cpuasm                      Check a Pep/9 microcode program for syntax errors. Supports 1- and 2-byte data buses as well as the extended control section detailed in Pep9Micro.
  cpurun                      Run a Pep/9 microcode program with an optional list of preconditions.
```
to
```
Name:
pep9term -- Translate and run Pep/9 assembly language and microcode programs.

Usage:
pep9term [--help] [--help-all] [--version] [--about] 
pep9term asm [--help] [-e error_file] -s source_file -o object_file 
pep9term run [--help] [-i charin_file] [-o charout_file] [--echo-output] [-m max_steps] -s object_file 
pep9term cpuasm [--help] [-e error_file] [--d2] -s microcode_file 
pep9term cpurun [--help] [-e error_file] [--d2] [-p precondition_file] -s microcode_file 


Flags:
  -h,--help     Show this help information.
  --help-all    Show help information for all subcommands.
  -v,--version  Display program version number.
  --about       Display information about licensing, Qt, and developers.

Subcommands:
  asm           Assemble a Pep/9 assembler source code program to object code.
  run           Run a Pep/9 object code program.
  cpuasm        Check a Pep/9 microcode program for syntax errors.
  cpurun        Run a Pep/9 microcode program with an optional list of preconditions.
```
The two changes are spacing of Flags description to be consistent with other mode `--help` outputs and the elimination of a feature list from `cpuasm`, which IMO is not appropriate here.

**Comment by Matthew-McRaven** [comment]
Are you indicating that you would like the subcommand & flag descriptions to be aligned like this example?
![aligned](https://user-images.githubusercontent.com/35609985/81621135-8135e780-93a2-11ea-8862-11a999211884.png)

**Comment by StanWarford** [comment]
No. More like this
```
Here is the result of running 'pep9term --help'
------------------------------------------------------------------------
Name:
pep9term -- Translate and run Pep/9 assembly language and microcode programs.

Usage:
pep9term [--help] [--help-all] [--version] [--about] 
pep9term asm [--help] [-e error_file] -s source_file -o object_file 
pep9term run [--help] [-i charin_file] [-o charout_file] [--echo-output] [-m max_steps] -s object_file 
pep9term cpuasm [--help] [-e error_file] [--d2] -s microcode_file 
pep9term cpurun [--help] [-e error_file] [--d2] [-p precondition_file] -s microcode_file 


Flags:
  -h,--help     Show this help information.
  --help-all    Show help information for all subcommands.
  -v,--version  Display program version number.
  --about       Display information about licensing, Qt, and developers.

Subcommands:
  asm           Assemble a Pep/9 assembler source code program to object code.
  run           Run a Pep/9 object code program.
  cpuasm        Check a Pep/9 microcode program for syntax errors.
  cpurun        Run a Pep/9 microcode program with an optional list of preconditions.
------------------------------------------------------------------------


Here is the result of running 'pep9term asm --help'
------------------------------------------------------------------------
Name:
pep9term asm -- Assemble a Pep/9 assembler source code program to object code.

Usage:
pep9term asm [--help] [--help-all] [-e] -s -o 


Flags:
  -h,--help       Show this help information.
  --help-all      Show help information for all subcommands.
  -e error_file   Override the name of the default error log file.
  -s source_file  Input Pep/9 source program for assembler.
  -o object_file  Output object code generated from source.
------------------------------------------------------------------------


Here is the result of running 'pep9term run --help'
------------------------------------------------------------------------
Name:
pep9term run -- Run a Pep/9 object code program.

Usage:
pep9term run [--help] [--help-all] [-i] [-o] [--echo-output] [-m] -s 


Flags:
  -h,--help        Show this help information.
  --help-all       Show help information for all subcommands.
  -i charin_file   File buffered behind the charIn input port.
  -o charout_file  File to which the charOut output port is streamed..
  --echo-output    Echo data written to charOut to std::out.
  -m max_steps     The maximum number of assembly instructions executed before aborting. Defaults to 25000
  -s object_file   Input Pep/9 object code program for simulator.
------------------------------------------------------------------------


Here is the result of running 'pep9term cpuasm --help'
------------------------------------------------------------------------
Name:
pep9term cpuasm -- Check a Pep/9 microcode program for syntax errors. Supports 1- and 2-byte data buses as well as the extended control section detailed in Pep9Micro.

Usage:
pep9term cpuasm [--help] [--help-all] [-e] [--d2] -s 


Flags:
  -h,--help          Show this help information.
  --help-all         Show help information for all subcommands.
  -e error_file      Override the name of the default error log file.
  --d2               Assemble the microcode program with a 2-byte data bus
  -s microcode_file  Input Pep/9 microcode source program for microassembler.
------------------------------------------------------------------------
```

**Comment by StanWarford** [comment]
Also, general usage, feature lists, and default values belong in a paragraph in the Name: item. Let me do a rightup and we can discuss.

**Comment by StanWarford** [comment]
Here is a diff run on my writeup vs current help output.

```
Stans-iMac-Pro:Downloads warford$ diff pep9term-help.txt pep9term-help-copy.txt
15,17c15,17
<   -h,--help     Show this help information.
<   -v,--version  Display program version number.
<   --about       Display information about licensing, Qt, and developers.
---
>   -h,--help                   Show this help information.
>   -v,--version                Display program version number.
>   --about                     Display information about licensing, Qt, and developers.
20,23c20,23
<   asm           Assemble a Pep/9 assembler source code program to object code.
<   run           Run a Pep/9 object code program.
<   cpuasm        Check a Pep/9 microcode program for syntax errors.
<   cpurun        Run a Pep/9 microcode program.
---
>   asm                         Assemble a Pep/9 assembler source code program to object code.
>   run                         Run a Pep/9 object code program.
>   cpuasm                      Check a Pep/9 microcode program for syntax errors.
>   cpurun                      Run a Pep/9 microcode program with an optional list of preconditions.
33c33
< pep9term asm [--help] [-e error_file] -s source_file -o object_file 
---
> asm [--help] [-e error_file] -s source_file -o object_file 
37d36
< 
39,42c38,41
<   -h,--help       Show this help information.
<   -e error_file   Override the name of the default error log file.
<   -s source_file  Input Pep/9 source program for assembler.
<   -o object_file  Output object code generated from source.
---
>   -h,--help                   Show this help information.
>   -e error_file               Override the name of the default error log file.
>   -s source_file              Input Pep/9 source program for assembler.
>   -o object_file              Output object code generated from source.
52c51
< pep9term run [--help] [-i charin_file] [-o charout_file] [--echo-output] [-m max_steps] -s object_file 
---
> run [--help] [-i charin_file] [-o charout_file] [--echo-output] [-m max_steps] -s object_file 
54c53
< The object_file must be a .pepo file. If the program takes input -i is required. If the program produces output -o is required. As a guard against endless loops the program will abort after max_steps assembly instructions execute. The default value of max_steps is 25000.
---
> The object_file must be a .pepo file. If the program takes input, -i is required. If the program produces output, -o is required. As a guard against endless loops the program will abort after max_steps assembly instructions execute. The default value of max_steps is 25000.
57,62c56,61
<   -h,--help        Show this help information.
<   -i charin_file   File buffered behind the charIn input port.
<   -o charout_file  File to which the charOut output port is streamed.
<   --echo-output    Echo data written to charOut to std::out.
<   -m max_steps     Override the default value of max_steps.
<   -s object_file   Input Pep/9 object code program for simulator.
---
>   -h,--help                   Show this help information.
>   -i charin_file              File buffered behind the charIn input port.
>   -o charout_file             File to which the charOut output port is streamed..
>   --echo-output               Echo data written to charOut to std::out.
>   -m max_steps                The maximum number of assembly instructions executed before aborting. Defaults to %1
>   -s object_file              Input Pep/9 object code program for simulator.
72,74c71
< pep9term cpuasm [--help] [-e error_file] [--d2] -s microcode_file 
< 
< The source_file must be a .pepcpu file. If there are micro-assembly errors an error log file named <source_file>_errLog.txt is created with the error messages. <source_file> is the name of source_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. Supports the extended control section detailed in Pep9Micro.
---
> cpuasm [--help] [-e error_file] [--d2] -s microcode_file 
75a73
> The microcode_file must be a .pepcpu file. If there are micro-assembly errors an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of microcode_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. Supports the extended control section detailed in Pep9Micro.
78,81c76,79
<   -h,--help          Show this help information.
<   -e error_file      Override the name of the default error log file.
<   --d2               Assemble the microcode program with a 2-byte data bus
<   -s microcode_file  Input Pep/9 microcode source program for microassembler.
---
>   -h,--help                   Show this help information.
>   -e error_file               Override the name of the default error log file.
>   --d2                        Assemble the microcode program with a 2-byte data bus
>   -s microcode_file           Input Pep/9 microcode source program for microassembler.
88c86
< pep9term cpurun -- Assemble and run a Pep/9 microcode program.
---
> pep9term cpurun -- Run a Pep/9 microcode program with an optional list of preconditions.
91c89
< pep9term cpurun [--help] [-e error_file] [--d2] [-p precondition_file] -s microcode_file 
---
> cpurun [--help] [-e error_file] [--d2] [-p precondition_file] -s microcode_file 
93c91
< The source_file must be a .pepcpu file. If there are micro-assembly errors or UnitPost errors an error log file named <source_file>_errLog.txt is created with the error messages. <source_file> is the name of source_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. If -p is specified then all UnitPre and UnitPost statements in source_file are ignored and the execution is tested with the unit test in precondition_file. The precondition_file must be a .pepcpu file.
---
> The microcode_file must be a .pepcpu file. If there are micro-assembly errors or UnitPost errors an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of source_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. If -p is specified then all UnitPre and UnitPost statements in microcode_file are ignored. The UnitPre and UnitPost statments from precondition_file will be used instead. The precondition_file must be a .pepcpu file.
94a93
> As a guard against endless loops the program will abort after max_steps assembly instructions execute. The default value of max_steps is %1.
97,101c96,100
<   -h,--help             Show this help information.
<   -e error_file         Override the name of the default error log file.
<   --d2                  Assemble and run the microcode program with a 2-byte data bus.
<   -p precondition_file  Use the unit test in precondition_file.
<   -s microcode_file     Input Pep/9 microcode source program for microassembler.
---
>   -h,--help                   Show this help information.
>   -e error_file               Override the name of the default error log file.
>   --d2                        Assemble and run the microcode program with a 2-byte data bus.
>   -p precondition_file        A Pep/9 .pepcpu file containg UnitPre and UnitPost statements. Using this flag overrides all UnitPre and UnitPost statements in source_file.
>   -s microcode_file           Input Pep/9 microcode source program for microassembler.
Stans-iMac-Pro:Downloads warford$ 
```

**Comment by Matthew-McRaven** [comment]
Some of those differences should be kept. For example, cprun and cpuasm do not have a "source_file" parameter, which is why it was changed to "microcode_file".

I cannot easily change the spacing between the flags and the corresponding help documentation.

I corrected the error where in sub-commands, it did not print out the full `pep9term cpurun ...`

**Comment by StanWarford** [comment]
OK. Here is my latest diff. Mostly missing periods and commas. Too bad you cannot include the file names under the Flags heading. I'm ok with the spacing. Can you include the name of the file in the Flags heading? For example
`  -e error_file          Override the name of the default error log file.
`
seems preferable to
`  -e                          Override the name of the default error log file.
`
```
23c23
<   cpurun                       Run a Pep/9 microcode program
---
>   cpurun                      Run a Pep/9 microcode program.
35c35
< The source_file must be a .pep file. The object_file must be a .pepo file. If there are assembly errors an error log file named <source_file>_errLog.txt is created with the error messages. <source_file> is the name of source_file without the .pep extension. If there are no errors the error log file is not created.
---
> The source_file must be a .pep file. The object_file must be a .pepo file. If there are assembly errors, an error log file named <source_file>_errLog.txt is created with the error messages. <source_file> is the name of source_file without the .pep extension. If there are no errors, the error log file is not created.
73c73
< The microcode_file must be a .pepcpu file. If there are micro-assembly errors an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of microcode_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. Supports the extended control section detailed in Pep9Micro.
---
> The microcode_file must be a .pepcpu file. If there are micro-assembly errors, an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of microcode_file without the .pepcpu extension. If there are no errors, the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. Supports the extended control section detailed in Pep9Micro.
78c78
<   --d2                        Assemble the microcode program with a 2-byte data bus
---
>   --d2                        Assemble the microcode program with a 2-byte data bus.
91c91
< The microcode_file must be a .pepcpu file. If there are micro-assembly errors or UnitPost errors an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of source_file without the .pepcpu extension. If there are no errors the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. If -p is specified then all UnitPre and UnitPost statements in microcode_file are ignored. The UnitPre and UnitPost statments from precondition_file will be used instead. The precondition_file must be a .pepcpu file.
---
> The microcode_file must be a .pepcpu file. If there are micro-assembly errors or UnitPost errors, an error log file named <microcode_file>_errLog.txt is created with the error messages. <microcode_file> is the name of source_file without the .pepcpu extension. If there are no errors. the error log file is not created. Supports 1- and 2-byte data buses with the 1-byte data bus as the default. If -p is specified, then all UnitPre and UnitPost statements in microcode_file are ignored. The UnitPre and UnitPost statments from precondition_file will be used instead. The precondition_file must be a .pepcpu file.
```

**Comment by Matthew-McRaven** [comment]
Latest commit adds the name of the flag parameter back.

I added the commas and missing periods.

**Comment by StanWarford** [comment]
Looks good! Except for one small space before an "R":

```
Subcommands:
  asm                         Assemble a Pep/9 assembler source code program to object code.
  run                         Run a Pep/9 object code program.
  cpuasm                      Check a Pep/9 microcode program for syntax errors.
  cpurun                       Run a Pep/9 microcode program.
```

**Comment by Matthew-McRaven** [comment]
Spacing fixed in 0e9ad340a7f2c7c57e5a7e577338aa54cb7f5547.

**Comment by StanWarford** [comment]
OK. Ready for release!
